tdiag/src/main.rs

```diff
@@ -91,29 +118,34 @@ You can customize the interface and port for the receiver (this program) with --
             println!("Trace sources connected");
             crate::commands::profile::listen_and_profile(timely_configuration, sockets)
         }
-        ("arrangements", Some(_args)) => {
-            println!(
-                "Listening for {} Timely connections on {}:{}",
-                source_peers, ip_addr, port
-            );
-            let timely_sockets =
-                tdiag_connect::receive::open_sockets(ip_addr.clone(), port, source_peers)?;
+        ("differential", Some(differential_args)) => {
```

Comment:
Perfect!

Reply:
Can you please split out the listener/socket setup (but not the command line parsing) into a helper function?

Reply:
Like so? I've moved it into `tdiag_connect`.